pkg/sqlmigrations/migrations.go

```diff
@@ -929,27 +887,49 @@ func upgradeTableDescsWithFn(
 				if err := kv.ValueProto(&sqlDesc); err != nil {
 					return err
 				}
-				if table := sqlDesc.GetTable(); table != nil {
-					if upgraded, err := upgradeFn(table); err != nil {
-						return err
-					} else if upgraded {
-						// Though SetUpVersion typically bails out if the table is being
-						// dropped, it's safe to ignore the DROP state here and
-						// unconditionally set UpVersion. For proof, see
-						// TestDropTableWhileUpgradingFormat.
-						//
-						// In fact, it's of the utmost importance that this migration
-						// upgrades every last old-format table descriptor, including those
-						// that are dropping. Otherwise, the user could upgrade to a version
-						// without support for reading the old format before the drop
-						// completes, leaving a broken table descriptor and the table's
-						// remaining data around forever. This isn't just a theoretical
-						// concern: consider that dropping a large table can take several
-						// days, while upgrading to a new version can take as little as a
-						// few minutes.
-						table.UpVersion = true
-						b.Put(kv.Key, sqlbase.WrapDescriptor(table))
+				switch t := sqlDesc.Union.(type) {
+				case *sqlbase.Descriptor_Table:
+					if table := sqlDesc.GetTable(); table != nil && upgradeTableDescFn != nil {
+						if upgraded, err := upgradeTableDescFn(table); err != nil {
+							return err
+						} else if upgraded {
+							// Though SetUpVersion typically bails out if the table is being
+							// dropped, it's safe to ignore the DROP state here and
+							// unconditionally set UpVersion. For proof, see
+							// TestDropTableWhileUpgradingFormat.
+							//
+							// In fact, it's of the utmost importance that this migration
+							// upgrades every last old-format table descriptor, including those
+							// that are dropping. Otherwise, the user could upgrade to a version
+							// without support for reading the old format before the drop
+							// completes, leaving a broken table descriptor and the table's
+							// remaining data around forever. This isn't just a theoretical
+							// concern: consider that dropping a large table can take several
+							// days, while upgrading to a new version can take as little as a
+							// few minutes.
+							table.UpVersion = true
+							if err := table.Validate(ctx, txn); err != nil {
```

Comment:
The `table.Validate` call is new. We really need to do that before writing a descriptor.